packages/api/src/controllers/v1/agit/mod.rs

```diff
@@ -81,14 +101,23 @@ impl AgitControllerV1 {
         Json(body): Json<AgitByIdAction>,
     ) -> Result<Json<Agit>> {
         tracing::debug!("agit act_by_id {id} {body:?}");
+
         match body {
-            AgitByIdAction::Update(_) => {
-                //TODO: Add Update Agit
-                Ok(Json(Agit::default()))
+            AgitByIdAction::Update(req) => {
+                let update_request = AgitRepositoryUpdateRequest {
+                    title: Some(req.title),
+                };
+
+                match ctrl.repo.update(id, update_request).await {
+                    Ok(updated_agit) => Ok(Json(updated_agit)),
+                    Err(e) => Err(ServiceError::Unknown(format!("Failed to update agit: {}", e).into())),
+                }
             }
             AgitByIdAction::Delete(_) => {
-                //TODO: Add Delete Agit
-                Ok(Json(Agit::default()))
+                match ctrl.repo.delete(id).await {
+                    Ok(_) => Ok(Json(Agit::default())), 
```

Comment:
```suggestion
                    Ok(agit) => Ok(Json(agit)), 
```

latest rust-sdk will return agit data when deleting data.
So update rust-sdk to origin/main, and use returned agit data.

And Also I recommend to separate this into a function

Reply:
okay, noted.

@hackartists